contrib/colab/DC-GAN.ipynb

```diff
@@ -0,0 +1,557 @@
+        "        self.bn2 = nn.BatchNorm1d(7*7*128) # reshape to (-1, 7, 7, 128)\n",
+        "        self.conv1 = nn.ConvTranspose2d(\n",
+        "            in_channels=128, \n",
+        "            out_channels=64, \n",
+        "            kernel_size=4,\n",
+        "            stride=2, \n",
+        "            padding=1, \n",
+        "            bias=False\n",
+        "        ) #relu activation\n",
+        "        self.bn3 = nn.BatchNorm2d(64)\n",
+        "        self.conv2 = nn.ConvTranspose2d(\n",
+        "            in_channels=64, \n",
+        "            out_channels=1, \n",
+        "            kernel_size=4,\n",
+        "            stride=2, \n",
+        "            padding=1, \n",
+        "            bias=False\n",
+        "        ) # tanh activation\n",
+        "              \n",
+        "    def forward(self, x):\n",
+        "        x = self.linear1(x)\n",
+        "        x = F.relu(x)\n",
+        "        x = self.bn1(x)\n",
+        "        x = self.linear2(x)\n",
+        "        x = F.relu(x)\n",
+        "        x = self.bn2(x)\n",
+        "        x = x.view(x.shape[0], 128, 7, 7)\n",
+        "        x = self.conv1(x)\n",
+        "        x = F.relu(x)\n",
+        "        x = self.bn3(x)\n",
+        "        x = self.conv2(x)\n",
+        "        x = torch.tanh(x)\n",
+        "        return x"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "3oBrhFUEQhxc",
+        "colab": {}
+      },
+      "source": [
+        "def init_weights(m):\n",
+        "    classname = m.__class__.__name__\n",
+        "    if classname.find('Conv') != -1 or classname.find('BatchNorm') != -1:\n",
+        "        m.weight.data.normal_(0.00, 0.02)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "EZ45r5UvQhxe",
+        "colab": {}
+      },
+      "source": [
+        "def real_data_target(size, device):\n",
+        "    '''\n",
+        "    Tensor containing ones, with shape = size\n",
+        "    '''\n",
+        "    data = torch.ones(size, 1)\n",
+        "    return data.to(device)\n",
+        "\n",
+        "def fake_data_target(size, device):\n",
+        "    '''\n",
+        "    Tensor containing zeros, with shape = size\n",
+        "    '''\n",
+        "    data = torch.zeros(size, 1)\n",
+        "    return data.to(device)"
+      ],
+      "execution_count": 0,
+      "outputs": []
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab_type": "code",
+        "id": "2ofS8wztQhxi",
+        "colab": {}
+      },
+      "source": [
+        "# Noise\n",
+        "def noise(size, device):\n",
+        "    n = torch.randn(size, 100)\n",
```

Comment:
can we make this into a constant, or a class variable in `GenerativeNet` ? This seems like a magic number but it's just the 1st dim of the 1st linear layer.

Reply:
Is is used at two places, first for feeding generative net and second to create inputs for test images. So it is more like general purpose function and hence kept outside.

Reply:
the code will fail if we change this to another number. It's therefore better to make this a constant, and use the constant in two places. It would also make sense if this was a method of the `GenerativeNet`, and you can use like `generator.generate_noise(size)`. up to you.

Reply:
Makes sense. thank you. done.